data/src/main/java/co/nimblehq/compose/crypto/data/repository/GlobalRepositoryImpl.kt

```diff
@@ -0,0 +1,53 @@
+package co.nimblehq.compose.crypto.data.repository
+
+import android.content.Context
+import android.net.*
+import co.nimblehq.compose.crypto.domain.repository.GlobalRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class GlobalRepositoryImpl(
+    context: Context
+) : GlobalRepository {
+
+    private var isNetworkConnected: MutableStateFlow<Boolean?> = MutableStateFlow(null)
```

Comment:
What if we set the `false` as the default value so we can get rid of null? If you agreed I think we can update and test it on integrate PR 🙏 
```suggestion
    private var isNetworkConnected: MutableStateFlow<Boolean> = MutableStateFlow(false)
```

Reply:
There is an issue in using `ConnectivityManager.NetworkCallback`. It doesn't recognize there is no internet at the start of an application in the following scenario

1. Turn off the internet
2. Open the app for the first time
3. The app will not show the dialog

Because `ConnectivityManager.NetworkCallback` only observes the internet changes after it has been initialized. So, `isNetworkConnected` can be `null` if the callback has not been initialized yet. 
There is a workaround for that but the purpose of this initiative is not about detecting the internet connection changes but to be able to show a global state of dialog so I would like to leave it as it is @Wadeewee. 🙏

Reply:
Ah, I see and Noted @kaungkhantsoe Thank for your explanation 🙇‍♀️